components/bl-calendar-heatmap/src/index.js

```diff
@@ -1,133 +1,173 @@
-    if (propMinCount === null) {
-      setMinCount(getMaxMinCalendarCount(newCalendarData).minCount);
-    } else {
-      setMinCount(propMinCount);
-    }
-  }, [propMinCount, newCalendarData]);
+  const maxValid = useMemo(() => data && Math.max(...data.map(({ value }) => value)), [data]);
+  const minValid = useMemo(() => data && Math.min(...data.map(({ value }) => value)), [data]);
 
   useEffect(() => {
-    if (ref.current && legendRef.current) {
-      handleResize();
-      window.addEventListener('resize', handleResize, false);
+    cal.paint({
+        range,
+        date: {
+          start    : startDate ? new Date(startDate) : new Date(),
+          min      : minDate,
+          max      : maxDate,
+          highlight: highlightDate && prepareHighlights(highlightDate),
+        },
+
+        data: {
+          source: sourceDataUrl || data,
+          type  : dataType,
+          x     : datePropName,
+          y     : valuePropName,
+          defaultDataValue,
+          groupY: groupYLogic.hasLogic
+            ? values => groupYLogic({ values })
+            : groupY,
+        },
+
+        scale: scaleValidate(
+          scaleColorRange, scaleColorScheme, scaleOpacityBaseColor,
+          scaleType, scaleDomain, minValid, maxValid
+        ),
+
+        domain: {
+          type,
+          dynamicDimension,
+          gutter,
+          sort,
+          label: {
+            text    : labelLogic.hasLogic
+              ? (timestamp, element) => labelLogic({ timestamp, element })
+              : label,
+            position: labelPosition,
+            rotate  : labelRotation,
+            textAlign,
+            offset  : {
+              x: labelOffsetX,
+              y: labelOffsetY,
+            },
+            height  : labelHeight,
+            width   : labelWidth,
+          },
+        },
+
+        subDomain: {
+          type  : subType,
+          sort  : subSort,
+          width : cellWidth,
+          height: cellHeight,
+          label : subLabelLogic.hasLogic
+            ? (timestamp, value, element) => subLabelLogic({ timestamp, value, element })
+            : subLabel,
+          color : subColorLabelLogic.hasLogic
+            ? (timestamp, value, backgroundColor) => subColorLabelLogic({ timestamp, value, backgroundColor })
+            : subColorLabel,
+          gutter: subGutter,
+          radius: cellRadius,
+        },
+
+        verticalOrientation,
+        animationDuration,
+        theme,
+        itemSelector: `#cal-heatmap-${ instanceId }`,
+      },
+      [
+        [Tooltip],
+        [
+          Legend,
+          {
+            enabled     : legend === 'Legend',
+            itemSelector: `#legend-label-${ instanceId }`,
+          },
+        ],
+        [
+          LegendLite,
+          {
+            enabled     : legend === 'LegendLite',
+            itemSelector: `#legend-label-${ instanceId }`,
+          },
+        ],
+        ...validateCalendarLabel(calendarLabel),
```

Comment:
need to make this logic more compact

You could add `shape<something>` functions.

shapeDomain / shapeSubdomain etc.